lib/Runtime/Library/JavascriptProxy.h

```diff
@@ -185,11 +185,13 @@ namespace Js
                     {
                         targetToTrapResultMap.Add(propertyId, true);
                     }
+                }
 
-                    if (fn(propertyRecord))
-                    {
-                        trapResult->DirectSetItemAt(trapResultIndex++, element);
-                    }
+                // We explcitly allow duplicates in the results. A map is sufficient since the spec steps that remove entries
```

Comment:
Test262 is run in nightlies and should sufficiently cover this case. I could add that one though..